pkg/pushsync/pushsync.go

```diff
@@ -380,12 +387,14 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllo
 			// if ErrWantSelf is returned, it means we are the closest peer.
 			return nil, fmt.Errorf("closest peer: %w", err)
 		}
-		if !ps.failedRequests.Useful(peer, ch.Address()) {
-			skipPeers = append(skipPeers, peer)
-			ps.metrics.TotalFailedCacheHits.Inc()
+
+		skipPeers = append(skipPeers, peer)
+
+		if ps.skipList.ShouldSkip(peer) {
```

Comment:
Why do we need the chunk map? I dont see it being used to make any decisions. We are just storing and getting stuff from it?

Reply:
We use the chunk map to check if a peer has been already skipped for that chunk. It's one way to make sure only one peer is skipped.